examples/demo_pyomo.py

```diff
@@ -0,0 +1,130 @@
+# -*- coding: utf-8 -*-
+try:
+    from pyomo.environ import *
+    from pyomo.opt import ProblemFormat
+except ImportError:
+    import sys
+    print "Pyomo is not installed."
+    sys.exit(0)
+
+from nlp.model.amplmodel import AmplModel
+from nlp.optimize.tron import TRON
+from nlp.optimize.pcg import TruncatedCG
+from nlp.tools.logs import config_logger
+import logging
+
+import numpy as np
+
+tron_logger = config_logger("nlp.tron",
+                            "%(name)-8s %(levelname)-5s %(message)s",
+                            level=logging.INFO)
+
+def create_simple_model():
+    model = ConcreteModel()
+    model.x = Var(bounds=(1., None))
+    model.o = Objective(expr=model.x)
+    model.x.set_value(20.0)
+    return model
+
+def create_rosenbrock_model():
+    model = ConcreteModel()
+    model.x = Var()
+    model.y = Var(bounds=(-1.5, None))
+    model.o = Objective(expr=(model.x-1)**2 + \
+                              100*(model.y-model.x**2)**2)
+    model.x.set_value(-2.)
+    model.y.set_value(1.)
+    return model
+
+def write_nl(model, nl_filename):
+    """Writes a Pyomo model in NL file format."""
+    _, smap_id = model.write(nl_filename,
```

Comment:
The `write_nl()` function doesn't really simplify anything. I suggest to remove it and call `model.write()` directly.